kjuntado/basic-app
# [Feature] Add Login view + Error Handling

## CHANGES
- Add login dialog
- Add fields for branch ID, username, and password
- Add icon for showing and hiding password
- Add login button that checks validity of fields using useValidator custom hook
- Add error message that shows current error
- Integrate with testData provided

## SCREENSHOTS
**First load**
![image](https://user-images.githubusercontent.com/105372569/172672991-5a583331-1749-47ce-8522-a590e836ac8d.png)

Error Message #1 
![image](https://user-images.githubusercontent.com/105372569/172673595-2b02e177-9fe4-4348-aed1-6b14d384010a.png)

Error Message #2
![image](https://user-images.githubusercontent.com/105372569/172673665-dacb77a6-06ae-49a4-b686-7b8f2a613100.png)

Error Message #3
![image](https://user-images.githubusercontent.com/105372569/172673739-d4d5cbc4-fd88-4618-959f-898f4574842f.png)

## TEST STEPS
Error Message #1:
1. Fill in only 1 field
2. Click **Login** button

Error Message #2:
1. Fill in all fields.
2. Set _Branch ID_ to a string
3. Click **Login** button

Error Message #3:
1. Fill in all fields. Make sure _Branch ID_ is a number
2. Click **Login** button

## NOTES
- Routing to follow
